detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/BuildFileFinder.java

```diff
@@ -0,0 +1,134 @@
+
+    public BuildFileFinder(final FileFinder fileFinder) {
+        this.fileFinder = fileFinder;
+    }
+
+    public File findTaskDependsFile(File sourceDir, File buildDir, boolean followSymLinks, Integer searchDepth) throws IntegrationException {
+        File taskDependsDotFile = fileFinder.findFile(buildDir, TASK_DEPENDS_FILE_NAME, followSymLinks, searchDepth);
+        if (taskDependsDotFile == null) {
+            logger.warn("Did not find {} in build dir {}; trying source dir", TASK_DEPENDS_FILE_NAME, buildDir.getAbsolutePath());
+            taskDependsDotFile = fileFinder.findFile(sourceDir, TASK_DEPENDS_FILE_NAME, followSymLinks, searchDepth);
+        }
+        if (taskDependsDotFile == null) {
+            throw new IntegrationException(String.format("Failed to find %s in either %s or %s",
+                TASK_DEPENDS_FILE_NAME, buildDir.getAbsolutePath(), sourceDir.getAbsolutePath()));
+        }
+        return taskDependsDotFile;
+    }
+
+    public File findLicenseManifestFile(File buildDir, String targetImageName, BitbakeEnvironment bitbakeEnvironment, boolean followSymLinks, int searchDepth) throws IntegrationException {
+        try {
+            String machineArch = bitbakeEnvironment.getMachineArch().orElse(null);
+            File licensesDir = findLicensesDir(buildDir, bitbakeEnvironment.getLicensesDirPath().orElse(null), followSymLinks, searchDepth);
+            logger.debug("Checking licenses dir {} for license.manifest for {}", licensesDir.getAbsolutePath(), targetImageName);
+            List<File> licensesDirContents = generateListOfFiles(licensesDir);
+            Optional<File> architectureSpecificManifestFile = findManifestFileForTargetArchitecture(targetImageName,
+                machineArch, licensesDirContents, followSymLinks);
+            if (architectureSpecificManifestFile.isPresent()) {
+                return architectureSpecificManifestFile.get();
+            }
+            logger.debug("Did not find a license.manifest for architecture {}; Will look for the most recent license.manifest file.", machineArch);
+            Optional<File> latestLicenseManifestFile = findMostRecentLicenseManifestFileForTarget(targetImageName, licensesDirContents, followSymLinks);
+            if (latestLicenseManifestFile.isPresent()) {
+                logger.debug("Found most recent license.manifest file: {}", latestLicenseManifestFile.get().getAbsolutePath());
+                return latestLicenseManifestFile.get();
+            }
+        } catch (Exception e) {
+            logger.debug(String.format("Error finding license.manifest file for target image %s", targetImageName), e);
+        }
+        throw new IntegrationException(String.format("Unable to find license.manifest file for target image %s", targetImageName));
+    }
+
+    @NotNull
+    private List<File> generateListOfFiles(final File licensesDir) {
+        // TODO surely there's a single-line way to do this (via nio or apache FileUtils)
```

Comment:
Walking the code in `FileUtils::listFiles` it appears they call `File::listFiles` under the hood and performs a null check.